code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -0,0 +1,207 @@
+        return data;
+    }
+
+    void setHost(ContainerEvent event, Instance instance) {
+        DataAccessor.fields(instance).withKey(InstanceConstants.FIELD_REQUESTED_HOST_ID)
+                .set(event.getHostId());
+    }
+
+    void setName(Map<String, Object> inspect, Instance instance) {
+        String name = (String)inspect.get(INSPECT_NAME);
+        name = name.replaceFirst("/", "");
+        instance.setName(name);
+    }
+
+    void setNetwork(Long accountId, Map<String, Object> inspect, Instance instance) {
+        String networkKind = NetworkConstants.KIND_NETWORK;
+        if ( BooleanUtils.toBoolean(getRancherNetworkLabel(inspect)) ) {
+            networkKind = NetworkConstants.KIND_HOSTONLY;
+        }
+
+        List<? extends Network> networks = networkDao.getNetworksForAccount(accountId, networkKind);
+        if ( networks.isEmpty() ) {
+            networks = networkDao.getNetworksForAccount(accountDao.getSystemAccount().getId(), networkKind);
+        }
+
+        Long networkId = !networks.isEmpty() ? networks.get(0).getId() : null;
+
+        if ( networkId != null ) {
+            List<Long> netIds = new ArrayList<Long>();
+            netIds.add(networkId);
+            DataAccessor.fields(instance).withKey(InstanceConstants.FIELD_NETWORK_IDS).set(netIds);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    void setImage(Map<String, Object> inspect, Instance instance) {
+        Map<String, Object> config = (Map<String, Object>)inspect.get(INSPECT_CONFIG);
+        String imageUuid = (String)config.get(INSPECT_IMAGE);
+
+        // Somewhat of a hack, but needed for testing against sim contexts
+        if ( !imageUuid.matches(IMAGE_KIND_PATTERN) ) {
+            imageUuid = IMAGE_PREFIX + imageUuid;
+        }
+        DataAccessor.fields(instance).withKey(InstanceConstants.FIELD_IMAGE_UUID).set(imageUuid);
+        Image image = null;
+        try {
+            image = storageService.registerRemoteImage(imageUuid);
+        } catch (IOException e) {
+            throw new ExecutionException("Unable to create image.", e);
+        }
+        instance.setImageId(image.getId());
+    }
+
+    String getRancherUuidLabel(Map<String, Object> inspect) {
+        return getLabel(RANCHER_UUID, inspect);
+    }
+
+    String getRancherNetworkLabel(Map<String, Object> inspect) {
+        return getLabel(RANCHER_NETWORK, inspect);
+    }
+
+    @SuppressWarnings("unchecked")
+    String getLabel(String key, Map<String, Object> inspect) {
+        try {
+            Map<String, Object> config = (Map<String, Object>)inspect.get(INSPECT_CONFIG);
+            List<String> envVars = (List<String>)config.get(INSPECT_ENV);
+            for ( String envVar : envVars ) {
+                if ( envVar.startsWith(key) ) {
+                    return envVar.substring(key.length());
+                }
+            }
```

Comment:
INSPECT_ENV can be legitimately null. I'll just change that to traditional null check.